modules/core/src/main/java/org/apache/ignite/internal/pagemem/wal/record/DataRecord.java

```diff
@@ -90,7 +83,31 @@ public DataRecord setWriteEntries(List<DataEntry> writeEntries) {
      * @return Collection of write entries.
      */
     public List<DataEntry> writeEntries() {
-        return writeEntries == null ? Collections.<DataEntry>emptyList() : writeEntries;
+        if (writeEntries instanceof DataEntry)
+            return Collections.singletonList((DataEntry)writeEntries);
+
+        return writeEntries == null ? Collections.emptyList() : (List<DataEntry>)writeEntries;
```

Comment:
I don't think that "writeEntries == null" is a realistic scenario. Other methods ignore such check. Can you please remove it here as well and maybe add "requireNotNull" in the constructor? Thank you!

Reply:
Fixed.